### PR TITLE
NN-5031 Spring Boot upgrade to 5.2.2 version

### PR DESCRIPTION
**Pull request by nagendramoj**
Main changes:

1. build.gradlet.kts - 
        a. Excluded `logging-classic` from the configuration to due to LoggerFactory issues.
        b. removed ehcahe reference
        c. repositories - packages added.
2. @Cacheable - getBasicKeyworkerDtoForStaffId & getBasicKeyworkerDtoForStaffId - Not required as suggested by @Michael Willis .
3. DomainEventListener & OffenderEventListener - Channged JMSListener to SQSListerner

Deleted Classes: Not required after the upgrade 
-----------------
1. CacheConfig.java - Configuration for ehcache and since we are not using any cache in keyworker-api so its redundant. 
2. CacheInfoContributor.java - Cache related class & also not referenced anywhere.
5. ClientTrackingConfigurationTest.kt - No longer required.
6. QueueHealthNegativeTest.kt - SQS Negative health check, as test cases are moved to another project and not required.
7. Logging.kt - Logging methods are not referenced.

### CODE REVIEW

File: src/test/resources/application-test.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ complexity_of_need_uri: http://localhost:10000
 spring:
   datasource:
     url: "jdbc:hsqldb:mem:keyworker-db;sql.syntax_pgs=true;shutdown=false"
+  flyway:
```
Comment (neilmendum) [inline src/test/resources/application-test.yml:35]: Same as https://github.com/ministryofjustice/hmpps-manage-adjudications-api/pull/330#discussion_r1162577187

**Review by neilmendum — COMMENTED** [review]
Think we also want to update a few other bits - will raise individual comments

**Comment by neilmendum** [comment]
`logback-spring.xml` - see https://github.com/ministryofjustice/hmpps-manage-adjudications-api/pull/330/files/8c18ec06ac71b8e3bb97c1c2176d441170db2e97#r1162579542

**Comment by neilmendum** [comment]
`ApplicationInsightsConfiguration` - see https://github.com/ministryofjustice/hmpps-manage-adjudications-api/pull/330/files/8c18ec06ac71b8e3bb97c1c2176d441170db2e97#r1162577694

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/security/ResourceServerConfiguration.java [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
 
-        http.sessionManagement()
-            .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-            .and().csrf().disable()// Can't have CSRF protection as requires session
-            .authorizeRequests(auth ->
-                auth.antMatchers("/webjars/**", "/favicon.ico", "/csrf",
-                        "/health/**", "/info", "/ping",
-                        "/v3/api-docs/**", "/v2/api-docs", "/swagger-ui/**", "/swagger-ui.html",
-                        "/swagger-resources", "/swagger-resources/configuration/ui", "/swagger-resources/configuration/security",
-                        "/queue-admin/retry-all-dlqs")
```
Comment (neilmendum) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/security/ResourceServerConfiguration.java:29]: Missed this from the paths we now pass to `httpRequest.requestMatchers`

File: src/test/resources/application-test.yml (hunk 1 above)
Comment (nagendramoj) [reply src/test/resources/application-test.yml:35]: No need to resolve, done already

**Comment by neilmendum** [comment]
`repositories` in `build.gradle.kts` - see https://github.com/ministryofjustice/hmpps-manage-adjudications-api/pull/330/files/8c18ec06ac71b8e3bb97c1c2176d441170db2e97#r1162577492

**Comment by neilmendum** [comment]
In `DomainEventListener` and `OffenderEventListener` we want to change `@JmsListener` to `@SqsListener` - see https://github.com/ministryofjustice/hmpps-incentives-api/pull/407/files#r1159682259

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/security/ResourceServerConfiguration.java (hunk 2 above)
Comment (nagendramoj) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/security/ResourceServerConfiguration.java:29]: Done ✅

File: src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -29,19 +29,19 @@
     </logger>
```
Comment (neilmendum) [inline src/main/resources/logback-spring.xml:30]: I think we also want to remove these sections
```xml
<springProfile name="dev || stdout">
<springProfile name="!(dev || stdout)">
```

For reference see https://github.com/ministryofjustice/hmpps-manage-adjudications-api/blob/8c18ec06ac71b8e3bb97c1c2176d441170db2e97/src/main/resources/logback-spring.xml

Comment (nagendramoj) [reply src/main/resources/logback-spring.xml:12]: Done ✅

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/ApplicationInsightsConfiguration.kt [hunk 4]
```diff
@@ -20,6 +20,8 @@ class ApplicationInsightsConfiguration {
     return TelemetryClient()
```
[the remark this answers is not in the document]
Comment (nagendramoj) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/config/ApplicationInsightsConfiguration.kt:18]: Done ✅

**Comment by neilmendum** [comment]
LGTM 🙌

**Review by neilmendum — APPROVED** [review]

**Pull request merged**